### PR TITLE
Update go.mod for controller-tools & controller-runtime

### PR DESCRIPTION
**Pull request by sgreene570**
https://github.com/kubernetes-sigs/controller-tools/pull/424 & https://github.com/kubernetes-sigs/controller-runtime/pull/836 have merged, so this PR updates the package versions accordingly. 

Interestingly enough, I ran into this when trying to configure my go environment. I was unable to locate `github.com/munnerz/controller-tools v0.1.10-0.20200323145043-a2d268fbf03d`.

Steps performed:

1. Remove replace directive for `controller-tools`.
2. Ran `go get sigs.k8s.io/controller-runtime@v0.6.0` (tag v0.6.0 contains PR 836)
3. Ran `go get sigs.k8s.io/controller-tools@v0.3.0` (tag v0.3.0 contains PR 424)
4. Ran `go mod tidy`
5. Ran `go mod vendor` & commited.

### CODE REVIEW

**Comment by sgreene570** [comment]
shrinks the vendor dir by ~.5mb, if anyone was wondering.

File: go.mod [hunk 1]
```diff
@@ -8,35 +8,26 @@ require (
...
 	k8s.io/api v0.18.3
 	k8s.io/apimachinery v0.18.3
-	k8s.io/apiserver v0.18.0
-	k8s.io/client-go v0.18.0
-
-	// Update when a tagged release includes https://github.com/kubernetes-sigs/controller-runtime/pull/836
-	sigs.k8s.io/controller-runtime v0.5.1-0.20200330174416-a11a908d91e0
-	sigs.k8s.io/controller-tools v0.2.2-0.20190919191502-76a25b63325a
+	k8s.io/apiserver v0.18.2
+	k8s.io/client-go v0.18.2
```
Comment (Miciah) [inline go.mod:30]: k8s.io/api is at v0.18.3 above; any idea why these aren't too?

Comment (sgreene570) [reply go.mod:30]: Ill set these to 0.18.3 as well so there's no longer a discrepancy. Thanks!

**Pull request merged**